### ISSUE TITLE
Accessing buffer's metadata

### ISSUE THREAD

**Issue by VRmnv** [go-gst/go-gst#106]
Hello everyone!

I'm trying to access buffer's metadata in appsink element. I want to extract SEIUser data and process it further. 

Example:

```
			buffer := sample.GetBuffer()

			buffer.ForEachMeta(func(meta *gst.Meta) bool {
				fmt.Println("metaType", meta.Info().Type().Name())
                                 // outputs: GstVideoSEIUserDataUnregisteredMeta
                                 ...
			})
```

I can use meta.Instance() to get C pointer for metadata, but i don't know how to unref it and transform to go []bytes to process them. How can i do this?

**Comment by RSWilli** [comment]
Hello, how would you do it in C?

**Comment by VRmnv** [comment]
I managed to convert GstVideoSEIUserDataUnregisteredMeta to GoBytes with C.GoBytes()

```
			buffer.ForEachMeta(func(meta *gst.Meta) bool {
				metaSize := int(meta.Info().Size())

				cMetaPtr := meta.Instance()

				// Convert the C pointer to a byte slice
				metaBytes := C.GoBytes(unsafe.Pointer(cMetaPtr), C.int(metaSize))
				log.Printf("metadata bytes: %v", metaBytes)

				return true
			})
```

**Comment by RSWilli** [comment]
I don't think this is the way to go though, you are converting the struct pointer and the metadata length to a byte slice. This works, but that doesn't mean that there is metadata in your slice. If you look at [the docs](https://gstreamer.freedesktop.org/documentation/video/gstvideosei.html?gi-language=c#GstVideoSEIUserDataUnregisteredMeta), you'll see the data that will be returned, in fact the slice will contain the struct members, which means pointers and length, instead of the actual data.

The bindings do not support `GstVideoSEIUserDataUnregisteredMeta` yet, but adding support for it shouldn't be too hard. If you could supply a reproduction pipeline that outputs these buffers that would be very helpful.

Lastly the C functions you would call if you were to do it in pure C would also be helpful.

If you'd like to tinker yourself in the bindings, go ahead. I'm happy to help. Please create a PR if you're finished.

**Comment by RSWilli** [comment]
this works now in the main branch with the generated bindings
